test/jdk/javax/swing/JTree/4314199/bug4314199.java

```diff
@@ -26,42 +26,46 @@
  * @test
  * @bug 4314199
  * @summary Tests that JTree repaints correctly in a container with a JMenu
- * @author Peter Zhelezniakov
- * @run applet/manual=yesno bug4314199.html
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual bug4314199
  */
 
-import javax.swing.*;
-import javax.swing.tree.*;
+import javax.swing.Box;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JTree;
+import javax.swing.UIManager;
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.DefaultTreeModel;
+import javax.swing.tree.TreePath;
 
-public class bug4314199 extends JApplet {
+public class bug4314199 {
 
-    public void init() {
+    private static final String INSTRUCTIONS = """
+            Select the last tree node (marked "Here") and click on the menu.
+            Look at the vertical line connecting nodes "Bug" and "Here".
+            If the connecting line does not disappear when the menu drops down,
+            press 'Pass' else 'Fail'. """;
 
-        try {
-            UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
-            SwingUtilities.invokeAndWait(new Runnable() {
+    private static JFrame frame;
```

Comment:
Frame can be moved inside creatUI(). If you can use splitUI instead of testUI that would be even better.